src/app/static/src/app/store/surveyAndProgram/actions.ts

```diff
@@ -188,23 +187,39 @@ export const actions: ActionTree<SurveyAndProgramState, RootState> & SurveyAndPr
 
     async validateSurveyAndProgramData(context) {
         const {commit} = context;
+        commit({type: SurveyAndProgramMutation.SelectedDataTypeUpdated, payload: null});
         await Promise.all(
             [
                 api<SurveyAndProgramMutation, SurveyAndProgramMutation>(context)
                     .withError(SurveyAndProgramMutation.SurveyError)
                     .withSuccess(SurveyAndProgramMutation.SurveyUpdated)
                     .freezeResponse()
-                    .get<SurveyResponse>("/disease/survey/"),
+                    .get<SurveyResponse>("/disease/survey/")
+                    .then((response) => {
+                    if (response) {
+                        commitSelectedDataTypeUpdated(commit, DataType.Survey);
```

Comment:
As per previous comment
>Ideally it would retain the previously selected data type if it was successfully validated, but would choose any other valid type if it wasn't (if any).

So I think rather than setting the value three times you might want to do something like push to an array of successful datatypes as you get  succesful responses. Then after all promises are complete, set selected data type to the previously selected type (if it's in the array) or to the first data type in the array (if there are any) if it's not